subworkflows/local/generate_genome.nf

```diff
@@ -3,46 +3,75 @@
 //
 // MODULE IMPORT BLOCK
 //
-include { CUSTOM_GETCHROMSIZES  } from '../../modules/nf-core/custom/getchromsizes/main'
-include { GNU_SORT              } from '../../modules/nf-core/gnu/sort'
-include { GET_LARGEST_SCAFF     } from '../../modules/local/get_largest_scaff'
+include { GET_LARGEST_SCAFF             } from '../../modules/local/get_largest_scaff'
+include { GENERATE_UNSORTED_GENOME      } from '../../subworkflows/local/generate_unsorted_genome'
+include { GENERATE_SORTED_GENOME        } from '../../subworkflows/local/generate_sorted_genome'
 
 workflow GENERATE_GENOME {
     take:
     reference_file  // Channel: path(file)
+    map_order       // Channel: val
 
     main:
     ch_versions     = Channel.empty()
+    ch_genomesize   = Channel.empty()
+    ch_genome_fai   = Channel.empty()
 
     //
     // MODULE: GENERATE INDEX OF REFERENCE
     //          EMITS REFERENCE INDEX FILE MODIFIED FOR SCAFF SIZES
     //
-    CUSTOM_GETCHROMSIZES (
-        reference_file,
-        "temp.genome"
+
+    reference_file
+        .combine(map_order)
+        .map{ ref_meta, ref, map_order ->
+             tuple(
+                [   id: ref_meta.id,
+                    map_order :map_order
+                ],
+                ref
+             )
+            }
+        .branch{
+            sorted      : it[0].map_order == "length"
```

Comment:
Making map_order a limited number of options in the future I think would be a good idea. This will stop people, that dont read the usage docs from inputting "size" and expecting it to work like "length". Or even inputting other stuff and expecting it to just work.

We can always work on this for a 1.2, so not a lbocker here, if needed but I do think it would be a good addition.

Reply:
I was considering the terminology too, and still need to find a good solution, also anticipate sorting by various of other parameters, alphabet, etc, which should be looked into, and will need a bigger brainstorm.